### DISCUSSION TITLE
Question: unused source files and classes

### DISCUSSION THREAD

**Discussion by vinjcc** [FunkinCrew/Funkin#1488] [Q&A]
while looking at the code, i noticed there are some files and classes that seem to go unused in the game. so i did this
```
cd source
for i in *; do echo $i; grep -n "\<${i%.hx}\>" *; done
```
then went ahead and made a short list:

+ `BeatBattle.hx`: nothing. just empty
+ `BlendModeEffect.hx` and `OverlayShader.hx`: for an unfinished red overlay effect for week 4
+ `ButtonRemapSubstate.hx`
+ `ChartParser.hx`: old code for putting notes together that is no longer used?
+ `ControlsSubState.hx`
+ `GameOverState.hx`: old game over screen from ludum dare prototype?
+ `LatencyState.hx`: some debug screen? no idea what this was for
+ `OptionsSubState.hx`

and
+ `WiggleEffect.hx`: this one i'm not sure about, `wiggleShit` (fun name) is mostly commented out in PlayState.hx and there's only the line `wiggleShit.update(Conductor.crochet);` at 2268 but i'm not sure if it has any effect.

which of these are to be implemented eventually, and which are fine to remove?

**Comment by fluttermsm** [comment]
you might be able to replace those source files with the current ones and compile?

**Comment by vinjcc** [comment]
it always compiled fine, the thing is most of these files can be removed entirely with no effect to the compilation process or the game itself. nm99's working on the options menu though which is nice.

**Comment by haya3218** [comment]
WiggleEffect.hx is used for an unused wiggle type effect in the thorns stage, later replaced with the animated wiggle effect seen in a spritesheet.
Re-enabling it works well and doesn't cost much frame drops.

**Comment by libauror-a** [comment]
BeatBattle.hx dosen't exist anymore due to my pull request so that's one file down.

**Comment by KitOdin** [comment]
you can just ignore them, or if your like me you can just add the unused states as easter eggs.
